assembl/static/js/app/views/messageColumnsPanel.js

```diff
@@ -0,0 +1,451 @@
+ */
+var BaseMessageColumnView = BaseMessageListMixin(Marionette.LayoutView);
+
+
+/**
+ * A single column of messages
+ * @class app.views.messageColumnsPanel.MessageColumnView
+ * @extends app.views.messageColumnsPanel.BaseMessageColumnView
+ */
+var MessageColumnView = BaseMessageColumnView.extend({
+  constructor: function MessageColumnView() {
+    BaseMessageColumnView.apply(this, arguments);
+  },
+  message_template: '#tmpl-messageColumn',
+
+  isCurrentViewStyleThreadedType: function() {
+    return false;
+  },
+  getTargetMessageViewStyleFromMessageListConfig: function() {
+    return Ctx.AVAILABLE_MESSAGE_VIEW_STYLES.FULL_BODY;
+  },
+  ui: {
+    panelBody: ".subpanel-body",
+    messageColumnHeader: '.js_messageColumnHeader',
+    messageColumnDescription: '.js_messageColumnDescription',
+    topPostRegion: '.js_topPostRegion',
+    messageFamilyList: '.js_messageFamilies_region',
+    pendingMessage: '.pendingMessage',
+    messageList: '.messageList-list',
+    topArea: '.js_messageList-toparea',
+    bottomArea: '.js_messageList-bottomarea',
+    contentPending: '.real-time-updates',
+    viewStyleDropdown: ".js_messageListViewStyle-dropdown",
+    messageCount: '.js_messageCount',
+  },
+  regions: {
+    messageFamilyList: '@ui.messageFamilyList',
+    topPostRegion: '@ui.topPostRegion',
+    messageColumnDescription: '@ui.messageColumnDescription',
+  },
+
+  /**
+   * This is a subset of {app.views.baseMessageList.BaseMessageListMixin.ViewStyles}
+   * @member app.views.messageColumnsPanel.ViewStyles
+   * @type {Object}
+   */
+  ViewStyles: {
+    REVERSE_CHRONOLOGICAL: {
+      id: "reverse_chronological",
+      css_class: MESSAGE_LIST_VIEW_STYLES_CLASS_PREFIX + "activityfeed",
+      label: i18n.gettext('Newest messages first')
+    },
+    POPULARITY: {
+      id: "popularity",
+      css_class: MESSAGE_LIST_VIEW_STYLES_CLASS_PREFIX + "popularmessages",
+      label: i18n.gettext('Most popular messages first')
+    },
+    CHRONOLOGICAL: {
+      id: "chronological",
+      css_class: MESSAGE_LIST_VIEW_STYLES_CLASS_PREFIX + "chronological",
+      label: i18n.gettext('Oldest messages first')
+    },
+  },
+
+  constrainViewStyle: function(viewStyle) {
+    if (!viewStyle) {
+      //If invalid, set global default
+      viewStyle = this.ViewStyles.REVERSE_CHRONOLOGICAL;
+    }
+    return viewStyle;
+  },
+
+  initialize: function(options) {
+    BaseMessageColumnView.prototype.initialize.apply(this, arguments);
+    var extraEvents = {},
+    collectionManager = new CollectionManager();
+    this.idea = options.idea;
+    this.showMessageByIdInProgress = false;
+    this.basePanel = options.basePanel;
+    this.translationDataPromise = options.translationDataPromise;
+    this.setViewStyle(this.ViewStyles.REVERSE_CHRONOLOGICAL);
+
+    _.each(this.ViewStyles, function(messageListViewStyle) {
+      var key = 'click .' + messageListViewStyle.css_class;
+      extraEvents[key] = 'onSelectMessageListViewStyle';
```

Comment:
Possible, yes. But we already inherit an events dictionary from BaseMessageColumnView, and so we'd have to combine. I thought that adding the new cases was simpler. But no fundamental objection if you prefer the other style.